Source/PluginProcessor.h

```diff
@@ -15,6 +15,163 @@
 #include "Tract.hpp"
 #include "WhiteNoise.hpp"
 #include "Biquad.hpp"
+#include <math.h>
+
+
+class PinkTromboneADSR
+{
+public:
+	//==============================================================================
+	PinkTromboneADSR()
+	{
+		recalculateRates();
+	}
+
+	//==============================================================================
+	struct Parameters
+	{
+		Parameters() = default;
+
+		Parameters (float attackTimeSeconds,
+					float decayTimeSeconds,
+					float sustainLevel,
+					float decayExponent)
+			: attack (attackTimeSeconds),
+			  decay (decayTimeSeconds),
+			  sustain (sustainLevel),
+			  decayExp(decayExponent)
+		{
+		}
+
+		float attack = 0.1f, decay = 0.1f, sustain = 1.0f, attackExp=1.0f, decayExp=1.0f;
+	};
+
+	void setParameters (const Parameters& newParameters)
+	{
+		// need to call setSampleRate() first!
+		jassert (sampleRate > 0.0);
+
+		parameters = newParameters;
+		recalculateRates();
+	}
+
+	//==============================================================================
+	void setSampleRate (double newSampleRate) noexcept
+	{
+		jassert (newSampleRate > 0.0);
+		sampleRate = newSampleRate;
+	}
+
+	//==============================================================================
+
+	void reset() noexcept
+	{
+		envelopeVal = 0.0f;
+		state = State::idle;
+	}
+	
+	void noteOn() noexcept
+	{
+		if (attackRate > 0.0f)
+		{
+			state = State::attack;
+		}
+		else if (decayRate > 0.0f)
+		{
+			envelopeVal = 1.0f;
+			state = State::decay;
+		}
+		else
+		{
+			envelopeVal = parameters.sustain;
+			state = State::sustain;
+		}
+	}
+	
+	void noteOff() noexcept
+	{
+		if (state != State::idle)
+			reset();
+	}
+
+	//==============================================================================
+	float getNextSample() noexcept
+	{
+		float scaledEnvelopeVal;
+		
+		if (state == State::idle)
+			return 0.0f;
+
+		if (state == State::attack)
+		{
+			envelopeVal += attackRate;
+
+			if (envelopeVal >= 1.0f)
+			{
+				envelopeVal = 1.0f;
+				goToNextState();
+			}
+		}
+		else if (state == State::decay)
+		{
+			envelopeVal -= decayRate;
+
+			if (envelopeVal <= parameters.sustain)
+			{
+				envelopeVal = parameters.sustain;
+				goToNextState();
+			}
+			if (parameters.decayExp >= 0)
+				scaledEnvelopeVal = powf(envelopeVal, parameters.decayExp);
+			else
+				scaledEnvelopeVal = powf(envelopeVal, (-1/parameters.decayExp));
+			return scaledEnvelopeVal;
+		}
+		else if (state == State::sustain)
+		{
+			envelopeVal = parameters.sustain;
+		}
+
+		return envelopeVal;
+	}
+
+private:
+	//==============================================================================
+	void recalculateRates() noexcept
+	{
+		auto getRate = [] (float distance, float timeInSeconds, double sr)

```

Comment:
Why exactly is this a local lambda? Seems like it might make more sense as a protected static class function.